examples/pbr/main.js

```diff
@@ -0,0 +1,1411 @@
+
+                    //                    group.addChild( this.testCubemapIrradiance( offsetX -60, 30 ) );
+
+                }
+
+
+                //group.getOrCreateStateSet().setAttributeAndModes( new osg.CullFace( 'DISABLE' ) );
+
+                // y up
+                osg.Matrix.makeRotate( -Math.PI / 2, -1, 0, 0, group.getMatrix() );
+
+                root.getOrCreateStateSet().addUniform( osg.Uniform.createInt( window.ROUGHNESS_TEXTURE_UNIT, 'roughnessMap' ) );
+                root.getOrCreateStateSet().addUniform( osg.Uniform.createInt( window.NORMAL_TEXTURE_UNIT, 'normalMap' ) );
+                root.getOrCreateStateSet().addUniform( osg.Uniform.createInt( window.METALNESS_TEXTURE_UNIT, 'specularMap' ) );
+                //                group.getOrCreateStateSet().addUniform( osg.Uniform.createInt( window.SPECULAR_TEXTURE_UNIT, 'specularMap' ) );
+                root.getOrCreateStateSet().addUniform( osg.Uniform.createInt( window.ALBEDO_TEXTURE_UNIT, 'albedoMap' ) );
+
+
+                this._viewer.getManipulator().computeHomePosition( true );
+                this.updateCameraPreset();
+
+
+            }.bind( this ) );
+
+            return root;
+        },
+
+        readEnvConfig: function ( file ) {
+
+            var d = P.defer();
+            var p = P.resolve( $.get( file ) );
+
+            p.then( function ( text ) {
+
+                var config = text;
+                d.resolve( config );
+
+            } );
+
+            return d.promise;
+        },
+
+        run: function ( canvas ) {
+
+            //osgGA.Manipulator.DEFAULT_SETTINGS = osgGA.Manipulator.DEFAULT_SETTINGS | osgGA.Manipulator.COMPUTE_HOME_USING_BBOX;
+            var viewer = this._viewer = new osgViewer.Viewer( canvas, {
+                preserveDrawingBuffer: true,
+                premultipliedAlpha: false
+            } );
+            viewer.init();
+
+            var gl = viewer.getState().getGraphicContext();
+            console.log( gl.getSupportedExtensions() );
+            console.log( gl.getExtension( 'OES_texture_float' ) );
+            var hasFloatLinear = gl.getExtension( 'OES_texture_float_linear' );
+            console.log( hasFloatLinear );
+            var hasTextureLod = gl.getExtension( 'EXT_shader_texture_lod' );
+            console.log( hasTextureLod );
+
+            var ready = [];
+
+            var environment;
+            // environment = 'textures/city_night_reference_2048/';
+            environment = 'textures/' + ( optionsURL.env ? optionsURL.env : 'sample_parking' ) + '/';
+
+
+            //var environment = 'textures/bus_garage5/';
+            //var environment = 'textures/walk_of_fame/';
+            //var environment = 'textures/airport/';
+            //var environment = 'textures/tmp/';
+
+            //var model = new ModelLoader( 'models/cerberus/' );
+
+            var modelPromises = [];
+            modelsPBR.forEach( function ( modelString, index ) {
```

Comment:
nop, we need to load only external pbr model, others a loaded differently. The var name should be externalModelPBR